src/vs/vscode.proposed.d.ts

```diff
@@ -33,7 +33,38 @@ declare module 'vscode' {
 		extensionHostEnv?: { [key: string]: string | null };
 	}
 
-	export type ResolverResult = ResolvedAuthority & ResolvedOptions;
+	export interface Port extends Disposable {
+		remotePort: number;
+		localPort?: number;
```

Comment:
Actually I think this should include remoteAddress (and possibly localAddress) too, not sure if that has been brought up.